lib/semian/circuit_breaker.rb

```diff
@@ -127,6 +133,10 @@ def log_state_transition(new_state)
       Semian.logger.info(str)
     end
 
+    def notify_state_transition(new_state)
+      Semian.notify(:state_change, self, nil, nil, state: new_state)
```

Comment:
I originally had that, but it requires the consumer to think about `nil` as the previous state (e.g. the initial transition to `closed`) which seemed unnecessary for my usecase. Always possible to add in later, with the `payload` system (rather than explicit parameters).